src/Button.php

```diff
@@ -12,81 +13,206 @@
  * For example,
  *
  * ```php
- * echo Button::widget()
- *     ->label('Action')
- *     ->options(['class' => 'btn-lg']);
+ * echo Button::widget()->label('Block button')->large()->type(ButtonType::PRIMARY)->render();
  * ```
  */
-final class Button extends Widget
+final class Button extends \Yiisoft\Widget\Widget
 {
-    /** @psalm-var non-empty-string */
-    private string $tagName = 'button';
+    private const NAME = 'btn';
+    private array $attributes = [];
+    private ButtonType $buttonType = ButtonType::SECONDARY;
+    private string|null $cssClass = null;
+    private bool|string $id = true;
     private string $label = 'Button';
-    private bool $encodeLabels = true;
-    private bool $encodeTags = false;
-    private array $options = [];
+    private string $tagName = 'button';
 
-    public function render(): string
+    /**
+     * Sets the button to be active.
+     *
+     * @return self A new instance of the current class with the button active.
+     */
+    public function active(): self
     {
-        if (!isset($this->options['id'])) {
-            $this->options['id'] = $this->getId();
-        }
+        $new = clone $this;
+        $new->attributes['aria-pressed'] = 'true';
+        $new->attributes['data-bs-toggle'] = 'button';
+        $new->cssClass = 'active';
 
-        if ($this->theme) {
-            $this->options['data-bs-theme'] = $this->theme;
-        }
+        return $new;
+    }
 
-        /** @psalm-suppress InvalidArgument */
-        Html::addCssClass($this->options, ['widget' => 'btn']);
+    /**
+     * Sets the CSS class attribute for the button component.
+     *
+     * @param string $value The CSS class for the button component (e.g., 'test-class').
+     *
+     * @return self A new instance of the current class with the specified class value.
+     *
+     * @link https://html.spec.whatwg.org/#classes
+     */
+    public function addClass(string $value): self
+    {
+        $new = clone $this;
+        $new->cssClass = $value;
```

Comment:
Should it add a class instead of overwriting it?

Reply:
We should add classes not overwrite them, good catch.

Reply:
done.